### PR TITLE
Ethclient

### PR DESCRIPTION
**Pull request by aaronbuchwald**
Provides solution for #46 . block.hash is producing an incorrect hash because some of the data is not present in the native Ethereum version of a block. Therefore, when it takes a hash of the block, it's missing part of the data and creating a different hash. Using this version of ethclient should resolve the issue.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -4,7 +4,9 @@ go 1.14
 
 require (
 	github.com/VictoriaMetrics/fastcache v1.5.7
+	github.com/ava-labs/avalanche-testing v0.9.6
 	github.com/ava-labs/avalanchego v1.0.4-update-id-2
+	github.com/ava-labs/gecko v0.6.1-rc.1
```
Comment (tasinco) [inline go.mod:9]: this should be avalanchego 1.0.5+ ?

Comment (aaronbuchwald) [reply go.mod:9]: Yes, good catch. Idk why this gets overrided sometimes when building.

File: go.mod [hunk 2]
```diff
@@ -23,6 +25,7 @@ require (
 	github.com/pborman/uuid v1.2.0
 	github.com/prometheus/tsdb v0.10.0
 	github.com/rjeczalik/notify v0.9.2
+	github.com/sirupsen/logrus v1.6.0
```
Comment (aaronbuchwald) [inline go.mod:28]: Need to remove this.

File: ethclient/ethclient.go [hunk 3]
```diff
@@ -0,0 +1,575 @@
...
+	var head *types.Header
+	var body rpcBlock
+	if err := json.Unmarshal(raw, &head); err != nil {
+		return nil, err
+	}
+	if err := json.Unmarshal(raw, &body); err != nil {
+		return nil, err
+	}
+	// Quick-verify transaction and uncle lists. This mostly helps with debugging the server.
+	if head.UncleHash == types.EmptyUncleHash && len(body.UncleHashes) > 0 {
```
Comment (aaronbuchwald) [inline ethclient/ethclient.go:125]: Remove uncle related calls

File: ethclient/ethclient.go [hunk 4]
```diff
@@ -0,0 +1,575 @@
...
+	StartingBlock hexutil.Uint64
+	CurrentBlock  hexutil.Uint64
+	HighestBlock  hexutil.Uint64
+	PulledStates  hexutil.Uint64
+	KnownStates   hexutil.Uint64
+}
+
+// SyncProgress retrieves the current progress of the sync algorithm. If there's
+// no sync currently running, it returns nil.
+func (ec *Client) SyncProgress(ctx context.Context) (*coreth.SyncProgress, error) {
```
Comment (aaronbuchwald) [inline ethclient/ethclient.go:327]: Remove if not implemented in coreth

**Pull request merged**